core/src/main/java/io/paradoxical/cassieq/dataAccess/QueueRepositoryImpl.java

```diff
@@ -29,95 +36,257 @@ public QueueRepositoryImpl(@NonNull final Session session) {
+
+        if (markedForDeletion) {
+            final Optional<DeletionJob> deletionJob = insertDeletionJobIfNotExists(definition);
+
+            if (tryAdvanceQueueStatus(definition.getQueueName(), QueueStatus.Deleting)) {
+                // anyone can create a queue with the same name now
+
+                logger.with("definition", definition).success("Marked for deletion");
+
+                return deletionJob;
+            }
+        }
 
-        initializeMonotonicValue(definition.getQueueName());
+        return Optional.empty();
+    }
+
+    private Optional<DeletionJob> insertDeletionJobIfNotExists(final QueueDefinition definition) {
+        final DeletionJob deletionJob = new DeletionJob(definition);
+
+        final Statement insert = QueryBuilder.insertInto(Tables.DeletionJob.TABLE_NAME)
+                                             .ifNotExists()
+                                             .value(Tables.DeletionJob.QUEUE_NAME, deletionJob.getQueueName().get())
+                                             .value(Tables.DeletionJob.VERSION, deletionJob.getVersion())
+                                             .value(Tables.DeletionJob.BUCKET_SIZE, deletionJob.getBucketSize().get());
+
+        if (session.execute(insert).wasApplied()) {
+            return Optional.of(deletionJob);
+        }
+
+        return Optional.empty();
+    }
 
-        initializePointers(definition.getQueueName());
+    @Override
+    public Optional<QueueDefinition> createQueue(@NonNull final QueueDefinition definition) {
+        if (upsertQueueDefinition(definition)) {
+            return getActiveQueue(definition.getQueueName());
+        }
 
-        insertQueueRecord(definition);
+        return Optional.empty();
     }
 
+    /**
+     * Set the status but only if its allowed to move to that status
+     *
+     * @param queueName
+     * @param status
+     */
     @Override
-    public void setQueueStatus(@NonNull final QueueName queueName, @NonNull final QueueStatus status) {
+    public boolean tryAdvanceQueueStatus(
+            @NonNull final QueueName queueName,
+            @NonNull final QueueStatus status) {
+
         final Statement update = QueryBuilder.update(Tables.Queue.TABLE_NAME)
-                                             .where(eq(Tables.Queue.QUEUENAME, queueName.get()))
-                                             .with(set(Tables.Queue.STATUS, status.name()));
+                                             .where(eq(Tables.Queue.QUEUE_NAME, queueName.get()))
+                                             .with(set(Tables.Queue.STATUS, status.ordinal()))
+                                             .onlyIf(lte(Tables.Queue.STATUS, status.ordinal()));
+
+        if (session.execute(update).wasApplied()) {
+            logger.with("queue-name", queueName)
+                  .with("new-status", status)
+                  .success("Advancing queue status");
+
+            return true;
+        }
+
+        return false;
```

Comment:
note, when i played with conditional upserts i noticed something really cool, if the upsert is NOT applied, it returns the current value of the condition columns. ... we COULD log those here....